lib/ActivityStreams.js

```diff
@@ -0,0 +1,144 @@
+/* globals NewTabURL, EventEmitter, XPCOMUtils */
+
+"use strict";
+
+const {Cu} = require("chrome");
+const {data} = require("sdk/self");
+const {PageMod} = require("sdk/page-mod");
+const {PlacesProvider} = require("lib/PlacesProvider");
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource:///modules/NewTabURL.jsm");
+
+XPCOMUtils.defineLazyGetter(this, "EventEmitter", function () {
+  const {EventEmitter} = Cu.import("resource://devtools/shared/event-emitter.js", {});
+  return EventEmitter;
+});
+
+let ActivityStreams = function (options) {
+  console.debug(`ActivityStreams.init`);
+
+  this.options = options ? options : {};
+  this.pageURL = this.options.pageURL ? options.pageURL : data.url("content/activity-streams.html");
+
+  this.workers = new Set();
```

Comment:
Maybe this needs to be a WeakSet, because we don't "own/create" the workers - they are handed to us `onAttach`.